daemon/src/model/cfd.rs

```diff
@@ -2361,6 +2373,140 @@ mod tests {
         assert_eq!(funding_fee.rate, funding_rate);
     }
 
+    #[test]
+    fn ensure_collaborative_settlement_takes_rollover_fees_into_account() {
+        let quantity = Usd::new(dec!(10));
+        let leverage = Leverage::new(2).unwrap();
+        let opening_price = Price::new(dec!(10000)).unwrap();
+        let closing_price = Price::new(dec!(10000)).unwrap();
+        let positive_funding_rate = dec!(0.0001);
+
+        assert_eq!(
+            calculate_long_margin(opening_price, quantity, leverage),
+            Amount::from_sat(50000)
+        );
+        assert_eq!(
+            calculate_short_margin(opening_price, quantity),
+            Amount::from_sat(100000)
+        );
+
+        // 0 rollover
+        assert_collab_settlement_taker_long_maker_short(
+            quantity,
+            leverage,
+            opening_price,
+            closing_price,
+            positive_funding_rate,
+            1000,
+            0,
+            Amount::from_sat(49583),
```

Comment:
Interestingly, if there is no price change at all and no fees we still end up in an interval that can be in our favor or not. I am not sure this can be avoided with a model the dicretizes evenly, but I never really took it into account when I thought about the values displayed in the UI.